### PR TITLE
Added ability to use comparison functions other than ==

### PR DESCRIPTION
**Pull request by Kheldar**

### CODE REVIEW

**Comment by jystic** [comment]
Thanks for the PR, will get a chance to look at this properly on the weekend.

**Comment by Kheldar** [comment]
Have you had a chance to look at this?

**Comment by jystic** [comment]
Sorry, not yet, I've been too busy, will get to it as soon as I can

**Comment by Kheldar** [comment]
Any chance this could be looked at?

File: hedgehog/src/Hedgehog/Internal/Property.hs [hunk 1]
```diff
@@ -545,13 +557,36 @@ infix 4 ===
...
-(===) x y = do
-  ok <- withFrozenCallStack $ eval (x == y)
+(===) = compareWith (==)
+
+infix 4 /==
+
+-- | Fails the test if the two arguments provided are equal.
+--
+(/==) :: (MonadTest m, Eq a, Show a, HasCallStack) => a -> a -> m ()
+(/==) = compareWith (/=)
```
Comment (puffnfresh) [inline hedgehog/src/Hedgehog/Internal/Property.hs]: This would do a diff when values are equal. Not ideal.

**Comment by jystic** [comment]
Diffs only make sense for equality like comparisons.

**Comment by jystic** [comment]
Similar functionality merged in https://github.com/hedgehogqa/haskell-hedgehog/pull/196

**Pull request closed without merging**